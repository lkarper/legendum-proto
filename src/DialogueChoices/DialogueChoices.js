import React from 'react';
import PropTypes from 'prop-types';
import DialogueResponse from '../DialogueResponse/DialogueResponse';
import './DialogueChoices.css';

const DialogueChoices = (props) => {
    const {
        dialogue, 
        page, 
        setResponse, 
    } = props;

    const { choices } = dialogue[page - 1];

    const choicesHTML = choices
        .split('|')
        .map((choice, i) => 
            <button 
                className='DialogueChoices_choice-button button'
                key={i} 
                onClick={() => setResponse(
                    <DialogueResponse 
                        dialogue={dialogue}
                        page={page}
                        choiceIndex={i}
                    />
                )}
            >
                {choice}
            </button>
        );

    if (!choices) {
        return (
            <div className='DialogueChoices__container'>
                <p>Error: Looks like something went wrong. Check your connection and try again.</p>
            </div>
        );        
    }

    return (
        <div className='DialogueChoices__container'>
            {choicesHTML}
        </div>
    );
}

DialogueChoices.defaultProps = {
    dialogue: [{
        choices: '',
    }], 
    page: 1, 
    setResponse: () => {},    
};

DialogueChoices.propTypes = {
    dialogue: PropTypes.arrayOf(
        PropTypes.shape({
            choices: PropTypes.string, 
        }),
    ),
};

export default DialogueChoices;
