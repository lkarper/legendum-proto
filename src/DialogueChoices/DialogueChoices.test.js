import React from 'react';
import ReactDOM from 'react-dom';
import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import DialogueChoices from './DialogueChoices';

describe('DialogueChoices component', () => {
    it('renders the UI without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<DialogueChoices />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the UI as expected', () => {
        const wrapper = mount(<DialogueChoices />);
        expect(toJSON(wrapper)).toMatchSnapshot();
        wrapper.unmount();
    });
});
