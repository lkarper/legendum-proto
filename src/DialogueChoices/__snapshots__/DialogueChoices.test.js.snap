// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DialogueChoices component renders the UI as expected 1`] = `
<DialogueChoices
  dialogue={
    Array [
      Object {
        "choices": "",
      },
    ]
  }
  page={1}
  setResponse={[Function]}
>
  <div
    className="DialogueChoices__container"
  >
    <p>
      Error: Looks like something went wrong. Check your connection and try again.
    </p>
  </div>
</DialogueChoices>
`;
